src/java.base/share/classes/java/lang/Thread.java

```diff
@@ -1656,19 +1686,15 @@ public void setContextClassLoader(ClassLoader cl) {
             security.checkPermission(
                 SecurityConstants.MODIFY_THREADGROUP_PERMISSION);
         }
-
-        // Get a snapshot of the list of all threads
-        Thread[] threads = getThreads();
-        StackTraceElement[][] traces = dumpThreads(threads);
-        Map<Thread, StackTraceElement[]> m = new HashMap<>(threads.length);
-        for (int i = 0; i < threads.length; i++) {
-            StackTraceElement[] stackTrace = traces[i];
-            if (stackTrace != null) {
-                m.put(threads[i], stackTrace);
-            }
-            // else terminated so we don't put it in the map
+        // Allow room for more Threads to be created before calling enumerate()
+        int count = systemThreadGroup.activeCount() + 20;
+        Thread[] threads = new Thread[count];
+        count = systemThreadGroup.enumerate(threads);
+        Map<Thread, StackTraceElement[]> result = new HashMap<>((count * 4) / 3);
```

Comment:
Why the `4/3` scaling here? I suggest this should just use `new HashMap<>(count)`; that class is in a better position to deal appropriately with the capacity request.

Reply:
This is current OpenJ9 behavour [1].
I think a change should be proposed to OpenJ9 head stream first, and ported to this afterwards.

[1] https://github.com/eclipse-openj9/openj9/blob/fc95e91365bc7a38f5abe0fd8b29b4672cb69a45/jcl/src/java.base/share/classes/java/lang/Thread.java#L1418

Reply:
Ok: I created https://github.com/eclipse-openj9/openj9/issues/14244.

Reply:
I'm fine with the change, we can just make it here and we'll get the same change into the head stream shortly.

Reply:
i.e. https://github.com/eclipse-openj9/openj9/pull/14245

Reply:
@keithc-ca any comments on https://github.com/eclipse-openj9/openj9/pull/14245#discussion_r782307818?

Reply:
See https://github.com/eclipse-openj9/openj9/pull/14245#discussion_r788124046.

Reply:
Ported `OpenJ9` change here.